pkg/metrics/tls.go

```diff
@@ -73,14 +74,19 @@ var _ verifier.TLSConfigurer = (*tlsConfigurerImpl)(nil)
 func newTLSConfigurer(certDir string, k8sClient kubernetes.Interface, clientCANamespace, clientCAConfigMap string) verifier.TLSConfigurer {
 	tlsRootConfig := verifier.DefaultTLSServerConfig(nil, nil)
 	tlsRootConfig.ClientAuth = tls.RequireAndVerifyClientCert
+	tlsConfigHolder := certwatch.NewTLSConfigHolder(tlsRootConfig, tls.RequireAndVerifyClientCert)
```

Comment:
I initially reused `tlsConfigHolder` else where and needed it as a separate variable, but removed that again. Then I liked it better as a separate variable because it can make debugging easier, so I kept it like that.